### PR TITLE
chore: Fixing non-propagated custom GH token in forked PRs 

### PR DESCRIPTION
**Pull request by heyvister1**
Since custom GH secrets are not propagated for forked repos, we need a way to have GH token to generate docs repo PRs under network-operator repo. 
Solution:

- Use fine grained Github Personal Access Token (PAT), for nvidia-ci-cd user , with limited network-operator-docs permissions

* There is a need to add `GH_DOCS_TOKEN` PAT token for `nvidia-ci-cd` user

### CODE REVIEW

**Comment by killianmuldoon** [comment]
What happens if some one creates a PR to this action with a script to exfiltrate the token before the check runs?

**Comment by heyvister1** [comment]
> What happens if some one creates a PR to this action with a script to exfiltrate the token before the check runs?

He needs to gain GH_TOKEN first, but he should be blocked on the first step which checks his membership

File: .github/workflows/docs-ci.yaml
Comment (maze88) [inline .github/workflows/docs-ci.yaml:42]: Does the GH API return an appropriate status code (`404`?) if the actor is not part of the organization? 
If so, this could be shortened to `if curl -f ... ; then`?

**Comment by killianmuldoon** [comment]
> He needs to gain GH_TOKEN first, but he should be blocked on the first step which checks his membership

What if I create an MR which changes that first step?

**Comment by heyvister1** [comment]
> > He needs to gain GH_TOKEN first, but he should be blocked on the first step which checks his membership
> 
> What if I create an MR which changes that first step?

So this is not only related to my PR change, if anyone can create its own workflow file which will be engaged in PR...
For current status, docs repo can be triggered manually by maintainers. So non-privileged collaborators cannot engage it.
The following workflow is triggered by network-operator PRs which is executing this workflow from main branch only. So others PRs theoretically should not affects our workflow security posture.

**Comment by maze88** [comment]
> > He needs to gain GH_TOKEN first, but he should be blocked on the first step which checks his membership
> 
> What if I create an MR which changes that first step?

The current trigger configuration doesn't trigger on PRs, so shouldn't this not be a problem?

And either way - we can configure the [repository (_Settings_ -> _Actions_ -> _General_ -> _Approval for running fork pull request workflows from contributors_](https://github.com/Mellanox/network-operator-docs/settings/actions) to _Require approval for all external contributors_.

**Review by e0ne — APPROVED** [review]

File: .github/workflows/docs-ci.yaml
Comment (adrianchiris) [inline .github/workflows/docs-ci.yaml:34]: i dont think this is needed.

this workflow is only triggered by other workflows not PRs.
the other workflow needs to have the needed ppermissions in order to trigger workflow in this repo.
currently only members of Mellanox org can trigger workflows for this project (i just modfied this in setting)

File: .github/workflows/docs-ci.yaml
Comment (adrianchiris) [reply .github/workflows/docs-ci.yaml:24]: need to make sure the GITHUB_TOKEN used in network-operator is sufficient.

File: .github/workflows/docs-ci.yaml
Comment (adrianchiris) [inline .github/workflows/docs-ci.yaml:24]: are we using this env var in the job ?

Comment (maze88) [reply .github/workflows/docs-ci.yaml:24]: @adrianchiris see my comment from the original line 33 in this file:
> # token must be explicitly set here for push to work in following step

@heyvister1 , although you renamed the secret variable ([in line 33](https://github.com/Mellanox/network-operator-docs/pull/134/files#diff-b1d6bf418157ab5fbc536f483dd7fae79a274c50824d70aae62de229c2023140R37)) - perhaps it's worth preserving the comment (unless you tested that after your change the token no longer needs to be explicitly set?).

File: .github/workflows/docs-ci.yaml
Comment (heyvister1) [inline .github/workflows/docs-ci.yaml:68]: No need to fork docs repo, since there is already an existing fork for nvidia-ci-cd user

**Pull request merged**
